wallet/internal/db/tx_pg.go

```diff
@@ -0,0 +1,272 @@
+		if params.Label != nil {
+			updateParams := sqlcpg.UpdateTransactionLabelParams{
+				TxHash:  params.Txid[:],
+				TxLabel: *params.Label,
+			}
+
+			err := qtx.UpdateTransactionLabel(ctx, updateParams)
+			if err != nil {
+				return fmt.Errorf("update transaction "+
+					"label: %w", err)
+			}
+		}
+
+		return nil
+	})
+}
+
+// GetTx retrieves a transaction record by its hash. Credits and debits are
+// derived from the UTXO store, not stored directly in the transaction record.
+func (w *PostgresWalletDB) GetTx(ctx context.Context,
+	query GetTxQuery) (*TxInfo, error) {
+
+	row, err := w.queries.GetTransactionByHash(ctx, query.Txid[:])
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, fmt.Errorf("%w: %s", ErrTxNotFound,
+				query.Txid)
+		}
+
+		return nil, fmt.Errorf("get transaction: %w", err)
+	}
+
+	return buildPgTxInfoFromGetRow(row)
+}
+
+// ListTxns returns a slice of transaction information based on the provided
+// query parameters.
+func (w *PostgresWalletDB) ListTxns(ctx context.Context,
+	query ListTxnsQuery) ([]TxInfo, error) {
+
+	if query.UnminedOnly {
+		// Get only unconfirmed transactions.
+		rows, err := w.queries.GetAllUnconfirmedTransactions(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("list unmined txs: %w", err)
+		}
+
+		txInfos := make([]TxInfo, 0, len(rows))
+		for _, row := range rows {
+			txInfo, err := buildTxInfoFromUnconfirmedPg(row)
+			if err != nil {
+				return nil, err
+			}
+
+			txInfos = append(txInfos, *txInfo)
+		}
+
+		return txInfos, nil
+	}
+
+	// Get transactions by height range.
+	startHeight, err := uint32ToInt32(query.StartHeight)
+	if err != nil {
+		return nil, fmt.Errorf("convert start height: %w", err)
+	}
+
+	endHeight, err := uint32ToInt32(query.EndHeight)
+	if err != nil {
+		return nil, fmt.Errorf("convert end height: %w", err)
+	}
+
+	params := sqlcpg.ListTransactionsByHeightRangeParams{
+		BlockHeight: sql.NullInt32{
+			Int32: startHeight,
+			Valid: true,
+		},
+		BlockHeight_2: sql.NullInt32{
+			Int32: endHeight,
+			Valid: true,
+		},
+	}
+
+	rows, err := w.queries.ListTransactionsByHeightRange(ctx, params)
+	if err != nil {
+		return nil, fmt.Errorf("list txs by height range: %w", err)
+	}
+
+	txInfos := make([]TxInfo, 0, len(rows))
+	for _, row := range rows {
+		txInfo, err := buildPgTxInfoFromListRow(row)
+		if err != nil {
+			return nil, err
+		}
+
+		txInfos = append(txInfos, *txInfo)
+	}
+
+	return txInfos, nil
+}
+
+// DeleteTx removes an unmined transaction from the store.
+func (w *PostgresWalletDB) DeleteTx(ctx context.Context,
+	params DeleteTxParams) error {
+
+	err := w.queries.DeleteUnconfirmedTransaction(ctx, params.Txid[:])
+	if err != nil {
+		return fmt.Errorf("delete unmined transaction: %w", err)
+	}
+
+	return nil
+}
+
+// RollbackToBlock removes all blocks at and after a given height, moving
+// transactions back to the unconfirmed pool.
+func (w *PostgresWalletDB) RollbackToBlock(ctx context.Context,
```

Comment:
We need to rethink about this method, maybe we could drop this method for the current PR - we also need to have the UTXOs table ready first before we can properly implement this method.